tests/ci/s3_helper.py

```diff
@@ -107,6 +107,9 @@ def _upload_file_to_s3(
         logging.info("Upload %s to %s. Meta: %s", file_path, url, metadata)
         return url
 
+    def delete_file_from_s3(self, bucket_name: str, s3_path: str) -> None:
+        self.client.delete_object(Bucket=bucket_name, Key=s3_path)
```

Comment:
It's not safe, right? So, it is not parallel-safe, AKA any job can fail in case of race conditions.

Reply:
```
In [21]: s3_helper.client.delete_object(Bucket='clickhouse-builds', Key='/asdasd/das/da/sd/asd/asd/a/sd/asd/as/d')
Out[21]:
{'ResponseMetadata': {'RequestId': '45J6KJ9C66JBWP00',
  'HostId': 'G/fbkzZMvKLVIcMQ71/8MBL0ND1wQI3+1vcgfXMXc4zf43N8mJZfffxA8qmzw5S27Af07U/IkqU=',
  'HTTPStatusCode': 204,
  'HTTPHeaders': {'x-amz-id-2': 'G/fbkzZMvKLVIcMQ71/8MBL0ND1wQI3+1vcgfXMXc4zf43N8mJZfffxA8qmzw5S27Af07U/IkqU=',
   'x-amz-request-id': '45J6KJ9C66JBWP00',
   'date': 'Wed, 31 Jan 2024 12:28:21 GMT',
   'server': 'AmazonS3'},
  'RetryAttempts': 0}}
```

It's safe